src/sentry/static/sentry/app/components/confirm.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-bootstrap/lib/Modal';
+
+import Button from './buttons/button';
+
+class Confirm extends React.PureComponent {
+  static propTypes = {
+    disabled: PropTypes.bool,
+    message: PropTypes.string.isRequired,
+    onConfirm: PropTypes.func.isRequired,
+    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
+    confirmText: PropTypes.string.isRequired,
+    cancelText: PropTypes.string.isRequired
+  };
+
+  static defaultProps = {
+    priority: 'primary',
+    cancelText: 'Cancel',
+    confirmText: 'Confirm'
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      isModalOpen: false,
+      confirming: false
+    };
+  }
+
+  handleConfirm = () => {
+    // `confirming` is used to make sure `onConfirm` is only called once
+    if (!this.state.confirming) {
```

Comment:
technically could this still double confirm if setState is async? :s (i'm not actually worried about this, just wondering for my knowledge

Reply:
Yeah, I had re-written it to use react state so that we could show a disabled button